CB-326: Hide and Unhide button for moderators on entity review page

code-master5 opened the pull request:
* If a review is not hidden, `Hide this review` button is shown to the moderator. Similarly, if the review is hidden, `Unhide this review` button is shown.
* A user cannot hide or unhide their review even if the user is a moderator. In this case, only some other moderator can take these actions (and is shown hide/unhide button).

paramsingh commented on critiquebrainz/frontend/templates/review/entity/base.html:
```diff
@@ -45,6 +45,25 @@ <h3>{{ review_credit(review, user_picture_size=24) }}</h3>
         {% endif %}
       {% endif %}
 
+      {% if current_user.is_authenticated and current_user != review.user and current_user.is_admin() %}
```
admins should be able to hide their own reviews as well. It won't be needed in real life, but it'd make testing easier.

code-master5 replied:
@paramsingh Ok.

code-master5 commented:
@paramsingh I've added some significant changes to this PR. Review now!

paramsingh commented:
>I've added some significant changes to this PR. 

A description of what the changes are is very helpful.

code-master5 commented:
@paramsingh sorry for not describing the changes. Changes are as follows -

* DB changes - added `unhide_review` and `unblock_user` into `action_types` enum.
* Defined `AdminActions` enum (python 3), instead of explicit variables in `db/moderation_log.py` module. Essentially, a python based enum is used to represent DB enum.
* Updated tests for `moderation_log` module.
* `review/unhide` and `user/unblock` endpoints now accept POST request to carry out DB
changes. This is done to keep the codebase in conformance with REST principles.
* Edited `review` and `user` templates to remove code redundancy. Also added cases for handling
`review unhide` and `user unblock` actions.
* Wrote test cases for `user` view almost from scratch, in order to make it more like other tests. Added tests for handling `user unblock` case.

paramsingh commented on admin/schema_changes/14.sql:
```diff
@@ -0,0 +1,2 @@
+ALTER TYPE action_types ADD VALUE 'unhide_review' AFTER 'hide_review';
```
This should be inside a transaction.
